---
layout: post
title: "ADAM 0.20.0 Released"
date: 2016-10-19 12:00:00 -0700
comments: true
categories: 
- ADAM
- releases
---

ADAM version 0.20.0 has been [released](https://github.com/bigdatagenomics/adam/releases)!

Due to major changes between Spark versions 1.6 and 2.0, we now build for combinations of Apache Spark and Scala versions:
[Spark 1.x and Scala 2.10](https://github.com/bigdatagenomics/adam/releases/tag/adam-parent_2.10-0.20.0),
[Spark 1.x and Scala 2.11](https://github.com/bigdatagenomics/adam/releases/tag/adam-parent_2.11-0.20.0),
[Spark 2.x and Scala 2.10](https://github.com/bigdatagenomics/adam/releases/tag/adam-parent-spark2_2.10-0.20.0), and
[Spark 2.x and Scala 2.11](https://github.com/bigdatagenomics/adam/releases/tag/adam-parent-spark2_2.11-0.20.0).

Since the last release, version 0.19.0, we have closed more than 180 issues and merged more than 120 pull requests.

We added a new pipe API, allowing for streaming alignment and variant records out to external applications and streaming back
in the results. Several new region join implementations are now public API, including a broadcast inner join, broadcast right
outer join, sort-merge inner join, sort-merge right outer join, sort-merge left outer join, sort-merge full outer join,
sort-merge inner join followed by a group by, and a sort-merge right outer join followed by a group by.

Alignment records can now be read from and written to CRAM format. We updated upstream dependencies on Hadoop-BAM and htsjdk to
fix various alignment record header bugs and to add support for gzip and BGZF compressed VCF.

Our sequence feature schema now more closely follow the GFF3 specification, while still supporting BED, GFF2/GTF, IntervalList,
and NarrowPeak formats. We also added a new sample schema for e.g. SRA sample metadata. 

With this version the core ADAM APIs are undergoing a major refactoring. We changed many method names on ADAMContext to make
the API more consistent. We also added RDD wrapper classes to increase performance by serializing metadata (such as record groups,
samples, and sequence dictionaries) to disk separate from primary data in Parquet. API incompatibilities between ADAM releases
will settle down by the 1.0 release, currently targeted for early 2017.

The full list of changes since version 0.19.0 is below.

<!-- more -->

**Closed issues:**

- Sorting by reference index seems doesn't work or sorted by DESC order? [\#1204](https://github.com/bigdatagenomics/adam/issues/1204)
- master won't compile [\#1200](https://github.com/bigdatagenomics/adam/issues/1200)
- VCF format tag SB field parse error in loading [\#1199](https://github.com/bigdatagenomics/adam/issues/1199)
- Publish sources JAR with snapshots [\#1195](https://github.com/bigdatagenomics/adam/issues/1195)
- Type SparkFunSuite in package org.bdgenomics.utils.misc is not available [\#1193](https://github.com/bigdatagenomics/adam/issues/1193)
- MDTagging fails on GRCh38 [\#1192](https://github.com/bigdatagenomics/adam/issues/1192)
- Fix stack overflow in IndelRealigner serialization [\#1190](https://github.com/bigdatagenomics/adam/issues/1190)
- Delete `./scripts/commit-pr.sh` [\#1188](https://github.com/bigdatagenomics/adam/issues/1188)
- Hadoop globStatus returns null if no glob matches [\#1186](https://github.com/bigdatagenomics/adam/issues/1186)
- Swapping out IntervalRDD under GenomicRDDs [\#1184](https://github.com/bigdatagenomics/adam/issues/1184)
- How to get "SO coordinate" instead of "SO unsorted"? [\#1182](https://github.com/bigdatagenomics/adam/issues/1182)
- How to read glob of multiple parquet Genotype [\#1179](https://github.com/bigdatagenomics/adam/issues/1179)
- Update command line doc and examples in README.md [\#1176](https://github.com/bigdatagenomics/adam/issues/1176)
- FastqRecordConverter needs cleanup and tests [\#1172](https://github.com/bigdatagenomics/adam/issues/1172)
- TransformFormats write to .gff3 file path incorrectly writes as parquet [\#1168](https://github.com/bigdatagenomics/adam/issues/1168)
- Should be able to merge shards across two different file systems [\#1165](https://github.com/bigdatagenomics/adam/issues/1165)
- RG ID gets written as the index, not the record group name [\#1162](https://github.com/bigdatagenomics/adam/issues/1162)
- Users should be able to save files as `-single` without merging them [\#1161](https://github.com/bigdatagenomics/adam/issues/1161)
- Users should be able to set size of buffer used for merging files [\#1160](https://github.com/bigdatagenomics/adam/issues/1160)
- Bump Hadoop-BAM to 7.7.0 [\#1158](https://github.com/bigdatagenomics/adam/issues/1158)
- adam-shell prints command trace to stdout [\#1154](https://github.com/bigdatagenomics/adam/issues/1154)
- Map IntervalList format column four to feature name or attributes? [\#1152](https://github.com/bigdatagenomics/adam/issues/1152)
- Parquet storage of VariantContext [\#1151](https://github.com/bigdatagenomics/adam/issues/1151)
- vcf2adam unparsable vcf record [\#1149](https://github.com/bigdatagenomics/adam/issues/1149)
- Reorder kryo.register statements in ADAMKryoRegistrator [\#1146](https://github.com/bigdatagenomics/adam/issues/1146)
- Make region joins public again [\#1143](https://github.com/bigdatagenomics/adam/issues/1143)
- Support CRAM input/output [\#1141](https://github.com/bigdatagenomics/adam/issues/1141)
- Transform should run with spark.kryo.requireRegistration=true [\#1136](https://github.com/bigdatagenomics/adam/issues/1136)
- adam-shell not handling bash args correctly [\#1132](https://github.com/bigdatagenomics/adam/issues/1132)
- Remove Gene and related models and parsing code [\#1129](https://github.com/bigdatagenomics/adam/issues/1129)
- Generate Scoverage reports when running CI [\#1124](https://github.com/bigdatagenomics/adam/issues/1124)
- Remove PairingRDD [\#1122](https://github.com/bigdatagenomics/adam/issues/1122)
- SAMRecordConverter.convert takes unused arguments [\#1113](https://github.com/bigdatagenomics/adam/issues/1113)
- Add Pipe API [\#1112](https://github.com/bigdatagenomics/adam/issues/1112)
- Improve coverage in Feature unit tests [\#1106](https://github.com/bigdatagenomics/adam/issues/1106)
- K-mer.scala code [\#1105](https://github.com/bigdatagenomics/adam/issues/1105)
- add -single file output option to ADAM2Vcf [\#1102](https://github.com/bigdatagenomics/adam/issues/1102)
- adam2vcf Fails with Sample not serializable [\#1100](https://github.com/bigdatagenomics/adam/issues/1100)
- ReferenceRegion.apply(AlignmentRecord) should not NPE on unmapped reads [\#1099](https://github.com/bigdatagenomics/adam/issues/1099)
- Add outer region join implementations [\#1098](https://github.com/bigdatagenomics/adam/issues/1098)
- VariantContextConverter never returns DatabaseVariantAnnotation [\#1097](https://github.com/bigdatagenomics/adam/issues/1097)
- loadvcf: conflicting require statement [\#1094](https://github.com/bigdatagenomics/adam/issues/1094)
- ADAM version 0.19.0 will not run on Spark version 2.0.0 [\#1093](https://github.com/bigdatagenomics/adam/issues/1093)
- Be more rigorous with FileSystem.get [\#1087](https://github.com/bigdatagenomics/adam/issues/1087)
- Remove network-connected and default test-related Maven profiles [\#1073](https://github.com/bigdatagenomics/adam/issues/1073)
- Releases should get pushed to Spark Packages [\#1067](https://github.com/bigdatagenomics/adam/issues/1067)
- Invalid POM for cli on 0.19.0 [\#1066](https://github.com/bigdatagenomics/adam/issues/1066)
- scala.MatchError RegExp does not catch colons in value part properly [\#1061](https://github.com/bigdatagenomics/adam/issues/1061)
- Support writing IntervalList header for features [\#1059](https://github.com/bigdatagenomics/adam/issues/1059)
- Add -single support when writing features in native formats [\#1058](https://github.com/bigdatagenomics/adam/issues/1058)
- Remove workaround for gzip/BGZF compressed VCF headers [\#1057](https://github.com/bigdatagenomics/adam/issues/1057)
- Clean up if clauses in Transform [\#1053](https://github.com/bigdatagenomics/adam/issues/1053)
- Adam-0.18.2 can not load Adam-0.14.0 adamSave function data (sam) [\#1050](https://github.com/bigdatagenomics/adam/issues/1050)
- filterByOverlappingRegion Incorrect for Genotypes [\#1042](https://github.com/bigdatagenomics/adam/issues/1042)
- Move Interval trait to utils, added in #75 [\#1041](https://github.com/bigdatagenomics/adam/issues/1041)
- Remove implicit GenomicRDD to RDD conversion [\#1040](https://github.com/bigdatagenomics/adam/issues/1040)
- VCF sample metadata - proposal for a GenotypedSampleMetadata object [\#1039](https://github.com/bigdatagenomics/adam/issues/1039)
- [build system] ADAM test builds pollute /tmp, leaving lots of cruft... [\#1038](https://github.com/bigdatagenomics/adam/issues/1038)
- adamMarkDuplicates function in AlignmentRecordRDDFunctions class can not mark the same read? [\#1037](https://github.com/bigdatagenomics/adam/issues/1037)
- test MarkDuplicatesSuite with two similar read in ref and start position and different avgPhredScore, error! [\#1035](https://github.com/bigdatagenomics/adam/issues/1035)
- Explore protocol buffers vs Avro [\#1031](https://github.com/bigdatagenomics/adam/issues/1031)
- Increase Avro dependency version to 1.8.0 [\#1029](https://github.com/bigdatagenomics/adam/issues/1029)
- ADAM specific logging [\#1024](https://github.com/bigdatagenomics/adam/issues/1024)
- Reenable Travis CI for pull request builds [\#1023](https://github.com/bigdatagenomics/adam/issues/1023)
- Bump Apache Spark version to 1.6.1 in Jenkins [\#1022](https://github.com/bigdatagenomics/adam/issues/1022)
- ADAM compatibility with Spark 2.0 [\#1021](https://github.com/bigdatagenomics/adam/issues/1021)
- ADAM to BAM conversion failing on 1000G file [\#1013](https://github.com/bigdatagenomics/adam/issues/1013)
- Factor out *RDDFunctions classes [\#1011](https://github.com/bigdatagenomics/adam/issues/1011)
- Port single file BAM and header code to VCF [\#1009](https://github.com/bigdatagenomics/adam/issues/1009)
- Roll Jenkins JDK 8 changes into ./scripts/jenkins-test [\#1008](https://github.com/bigdatagenomics/adam/issues/1008)
- Support GFF3 format [\#1007](https://github.com/bigdatagenomics/adam/issues/1007)
- Separate fat jar build from adam-cli to new maven module [\#1006](https://github.com/bigdatagenomics/adam/issues/1006)
- adam-cli POM invalid: maven.build.timestamp [\#1004](https://github.com/bigdatagenomics/adam/issues/1004)
- Sub-partitioning of Parquet file for ADAM [\#1003](https://github.com/bigdatagenomics/adam/issues/1003)
- Flattening the Genotype schema [\#1002](https://github.com/bigdatagenomics/adam/issues/1002)
- install adam 0.19 error! [\#1001](https://github.com/bigdatagenomics/adam/issues/1001)
- How to solve it please? [\#1000](https://github.com/bigdatagenomics/adam/issues/1000)
- Has the project realized alignment reads to reference genome algorithm? [\#996](https://github.com/bigdatagenomics/adam/issues/996)
- All file-based input methods should support running on directories, compressed files, and wildcards [\#993](https://github.com/bigdatagenomics/adam/issues/993)
- Contig to ContigName Change not reflected in AlignmentRecordField [\#991](https://github.com/bigdatagenomics/adam/issues/991)
- Add homebrew guidelines to release checklist or automate PR generation [\#987](https://github.com/bigdatagenomics/adam/issues/987)
- fix deprecation warnings [\#985](https://github.com/bigdatagenomics/adam/issues/985)
- rename `fragments` package [\#984](https://github.com/bigdatagenomics/adam/issues/984)
- Explore if SeqDict data can be factored out more aggressively [\#983](https://github.com/bigdatagenomics/adam/issues/983)
- Make "Adam" all caps in filename Adam2Fastq.scala [\#981](https://github.com/bigdatagenomics/adam/issues/981)
- Adam2Fastq should output reverse complement when 0x10 flag is set for read [\#980](https://github.com/bigdatagenomics/adam/issues/980)
- Allow lowercase letters in jar/version names [\#974](https://github.com/bigdatagenomics/adam/issues/974)
- Add stringency parameter to flagstat [\#973](https://github.com/bigdatagenomics/adam/issues/973)
- Arg-array parsing problem in adam-submit [\#971](https://github.com/bigdatagenomics/adam/issues/971)
- Pass recordGroup parameter to loadPairedFastq [\#969](https://github.com/bigdatagenomics/adam/issues/969)
- Send a number of partitions to sc.textFile calls [\#968](https://github.com/bigdatagenomics/adam/issues/968)
- adamGetReferenceString doesn't reduce pairs correctly [\#967](https://github.com/bigdatagenomics/adam/issues/967)
- Update ADAM formula in homebrew-science to version 0.19.0 [\#963](https://github.com/bigdatagenomics/adam/issues/963)
- BAM output in ADAM appears to be corrupt [\#962](https://github.com/bigdatagenomics/adam/issues/962)
- Remove code workarounds necessary for Spark 1.2.1/Hadoop 1.0.x support [\#959](https://github.com/bigdatagenomics/adam/issues/959)
- Issue with version 18.0.2 [\#957](https://github.com/bigdatagenomics/adam/issues/957)
- Expose sorting by reference index [\#952](https://github.com/bigdatagenomics/adam/issues/952)
- .rgdict and .seqdict files are not placed in the adam directory [\#945](https://github.com/bigdatagenomics/adam/issues/945)
- Why does count_kmers not return k-mers that are split between two records? [\#930](https://github.com/bigdatagenomics/adam/issues/930)
- Load legacy file formats to Spark SQL Dataframes [\#912](https://github.com/bigdatagenomics/adam/issues/912)
- Clean up RDD method names [\#910](https://github.com/bigdatagenomics/adam/issues/910)
- Load/store sequence dictionaries alongside Genotype RDDs [\#909](https://github.com/bigdatagenomics/adam/issues/909)
- vcf2adam -print_metrics throws IllegalStateException on Spark 1.5.2 or later [\#902](https://github.com/bigdatagenomics/adam/issues/902)
- error: no reads in first split: bad BAM file or tiny split size? [\#896](https://github.com/bigdatagenomics/adam/issues/896)
- FastaConverter.FastaDescriptionLine not kryo-registered [\#893](https://github.com/bigdatagenomics/adam/issues/893)
- Work With ADAM fasta2adam in a distributed mode [\#881](https://github.com/bigdatagenomics/adam/issues/881)
- vcf2adam -> Exception in thread "main" java.lang.NoSuchMethodError: scala.Predef$.$conforms()Lscala/Predef$$less$colon$less; [\#871](https://github.com/bigdatagenomics/adam/issues/871)
- Code coverage profile is broken [\#849](https://github.com/bigdatagenomics/adam/issues/849)
- Building Adam on OS X 10.10.5 with Java 1.8 [\#835](https://github.com/bigdatagenomics/adam/issues/835)
- Normalize AlignmentRecord.recordGroup* fields onto a separate record type [\#828](https://github.com/bigdatagenomics/adam/issues/828)
- Gracefully handle missing Spark- and Hadoop-versions in jenkins-test; document how to set them. [\#827](https://github.com/bigdatagenomics/adam/issues/827)
- Use Adam File with Hive [\#820](https://github.com/bigdatagenomics/adam/issues/820)
- How do we handle reads that don't have original quality scores when converting to FASTQ with original qualities? [\#818](https://github.com/bigdatagenomics/adam/issues/818)
- SAMFileHeader "sort order" attribute being un-set during file-save job [\#800](https://github.com/bigdatagenomics/adam/issues/800)
- Use same sort order as Samtools [\#796](https://github.com/bigdatagenomics/adam/issues/796)
- RNAME and RNEXT fields jumbled on transform BAM->ADAM->BAM [\#795](https://github.com/bigdatagenomics/adam/issues/795)
- Support loading multiple indexed read files [\#787](https://github.com/bigdatagenomics/adam/issues/787)
- Duplicate OUTPUT command line argument metaVar in adam2fastq [\#776](https://github.com/bigdatagenomics/adam/issues/776)
- Allow Variant to ReferenceRegion conversion [\#768](https://github.com/bigdatagenomics/adam/issues/768)
- Spark Errors References Deprecated SPARK_CLASSPATH [\#767](https://github.com/bigdatagenomics/adam/issues/767)
- Spark Errors References Deprecated SPARK_CLASSPATH [\#766](https://github.com/bigdatagenomics/adam/issues/766)
- adam2vcf fails with -coalesce [\#735](https://github.com/bigdatagenomics/adam/issues/735)
- Writing to a BAM file with adamSAMSave consistently fails [\#721](https://github.com/bigdatagenomics/adam/issues/721)
- BQSR on C835.HCC1143_BL.4 uses excessive amount of driver memory [\#714](https://github.com/bigdatagenomics/adam/issues/714)
- Support writing RDD[Feature] to various file formats [\#710](https://github.com/bigdatagenomics/adam/issues/710)
- adamParquetSave has a menacing false error message about *.adam extension [\#681](https://github.com/bigdatagenomics/adam/issues/681)
- BAMHeader not set when running on a cluster [\#676](https://github.com/bigdatagenomics/adam/issues/676)
- spark 1.3.1 upgarde to hortonworks HDP 2.2.4.2-2? [\#675](https://github.com/bigdatagenomics/adam/issues/675)
- `Symbol` case class is nucleotide-centric [\#672](https://github.com/bigdatagenomics/adam/issues/672)
- xAssembler cannot be build using mvn [\#658](https://github.com/bigdatagenomics/adam/issues/658)
- adam-submit VerifyError [\#642](https://github.com/bigdatagenomics/adam/issues/642)
- vcf2adam : Unsupported type ENUM [\#638](https://github.com/bigdatagenomics/adam/issues/638)
- Update CDH documentation [\#615](https://github.com/bigdatagenomics/adam/issues/615)
- Remove and generalize plugin code [\#602](https://github.com/bigdatagenomics/adam/issues/602)
- Fix record oriented shuffle [\#599](https://github.com/bigdatagenomics/adam/issues/599)
- Migrate preprocessing stages out of ADAM [\#598](https://github.com/bigdatagenomics/adam/issues/598)
- Publish/socialize a roadmap [\#591](https://github.com/bigdatagenomics/adam/issues/591)
- Eliminate format detection and extension checks for loading data [\#587](https://github.com/bigdatagenomics/adam/issues/587)
- Improve error message when we can't find a ReferenceRegion for a contig [\#582](https://github.com/bigdatagenomics/adam/issues/582)
- Do reference partitioners restrict a partition to contain keys from a single contig? [\#573](https://github.com/bigdatagenomics/adam/issues/573)
- Connection refused errors when transforming BAM file with BQSR [\#516](https://github.com/bigdatagenomics/adam/issues/516)
- ReferenceRegion shouldn't extend Ordered [\#511](https://github.com/bigdatagenomics/adam/issues/511)
- Documentation for common usecases [\#491](https://github.com/bigdatagenomics/adam/issues/491)
- Improve handling of "*" sequences during BQSR [\#484](https://github.com/bigdatagenomics/adam/issues/484)
- Original qualities are parsed out, but left in attribute fields [\#483](https://github.com/bigdatagenomics/adam/issues/483)
- Need a FileLocator that mirrors the use of Path in HDFS [\#477](https://github.com/bigdatagenomics/adam/issues/477)
- FileLocator should support finding "child" locators. [\#476](https://github.com/bigdatagenomics/adam/issues/476)
- Add S3 based Parquet directory loader [\#463](https://github.com/bigdatagenomics/adam/issues/463)
- Should FASTQ output use reads' "original qualities"? [\#436](https://github.com/bigdatagenomics/adam/issues/436)
- VcfStringUtils unused? [\#428](https://github.com/bigdatagenomics/adam/issues/428)
- We should be able to filter genotypes that overlap a region [\#422](https://github.com/bigdatagenomics/adam/issues/422)
- Create a simplified vocabulary for naming projections. [\#419](https://github.com/bigdatagenomics/adam/issues/419)
- Update documentation [\#406](https://github.com/bigdatagenomics/adam/issues/406)
- Bake off different region join implementations [\#395](https://github.com/bigdatagenomics/adam/issues/395)
- Handle no-ops more intelligently when creating MD tags [\#392](https://github.com/bigdatagenomics/adam/issues/392)
- Remove all the commands in the "CONVERSION OPERATIONS" `CommandGroup` [\#373](https://github.com/bigdatagenomics/adam/issues/373)
- Fail to Write RDD into HDFS with Parquet Format [\#344](https://github.com/bigdatagenomics/adam/issues/344)
- Refactor ReferencePositionWithOrientation [\#317](https://github.com/bigdatagenomics/adam/issues/317)
- Add docs about SPARK_LOCAL_IP [\#305](https://github.com/bigdatagenomics/adam/issues/305)
- PartitionAndJoin should throw an exception if it sees an unmapped read [\#297](https://github.com/bigdatagenomics/adam/issues/297)
- Add insert size calculation [\#296](https://github.com/bigdatagenomics/adam/issues/296)
- Newbie questions - learning resources? Reading a range of records from Adam? [\#281](https://github.com/bigdatagenomics/adam/issues/281)
- Add variant effect ontology [\#261](https://github.com/bigdatagenomics/adam/issues/261)
- Don't flatten optional SAM tags into a string [\#240](https://github.com/bigdatagenomics/adam/issues/240)
- Characterize impact of partition size on pileup creation [\#163](https://github.com/bigdatagenomics/adam/issues/163)
- Need to support BCF output format [\#153](https://github.com/bigdatagenomics/adam/issues/153)
- Allow list of commands to be injected into adam-cli AdamMain [\#132](https://github.com/bigdatagenomics/adam/issues/132)
- Parse out common annotations stored in VCF format [\#118](https://github.com/bigdatagenomics/adam/issues/118)
- Update normalization code to enable normalization of sequences with more than two indels [\#64](https://github.com/bigdatagenomics/adam/issues/64)
- Add clipping heuristic to indel realigner [\#63](https://github.com/bigdatagenomics/adam/issues/63)
- BQSR should support recalibration across multiple ADAM files [\#58](https://github.com/bigdatagenomics/adam/issues/58)

**Merged and closed pull requests:**

- fix SB tag parsing [\#1209](https://github.com/bigdatagenomics/adam/pull/1209) ([fnothaft](https://github.com/fnothaft))
- Fastq record converter [\#1208](https://github.com/bigdatagenomics/adam/pull/1208) ([fnothaft](https://github.com/fnothaft))
- Doc suggested partitionSize in ShuffleRegionJoin [\#1207](https://github.com/bigdatagenomics/adam/pull/1207) ([jpdna](https://github.com/jpdna))
- Test demonstrating region join failure [\#1206](https://github.com/bigdatagenomics/adam/pull/1206) ([jpdna](https://github.com/jpdna))
- fix SB tag parsing [\#1203](https://github.com/bigdatagenomics/adam/pull/1203) ([jpdna](https://github.com/jpdna))
- fix build [\#1201](https://github.com/bigdatagenomics/adam/pull/1201) ([ryan-williams](https://github.com/ryan-williams))
- \[ADAM-1192\] Correctly handle other whitespace in FASTA description. [\#1198](https://github.com/bigdatagenomics/adam/pull/1198) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1190\] Manually (un)pack IndelRealignmentTarget set. [\#1191](https://github.com/bigdatagenomics/adam/pull/1191) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1188\] Delete scripts/commit-pr.sh [\#1189](https://github.com/bigdatagenomics/adam/pull/1189) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1186\] Mask null from fs.globStatus. [\#1187](https://github.com/bigdatagenomics/adam/pull/1187) ([fnothaft](https://github.com/fnothaft))
- Fastq record converter [\#1185](https://github.com/bigdatagenomics/adam/pull/1185) ([zyxue](https://github.com/zyxue))
- \[ADAM-1182\] isSorted=true should write SO:coordinate in SAM/BAM/CRAM header. [\#1183](https://github.com/bigdatagenomics/adam/pull/1183) ([fnothaft](https://github.com/fnothaft))
- Add scoverage aggregator and fail on low coverage. [\#1181](https://github.com/bigdatagenomics/adam/pull/1181) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1179\] Improve error message when globbing a parquet file fails. [\#1180](https://github.com/bigdatagenomics/adam/pull/1180) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1176\] Update command line doc and examples in README.md [\#1177](https://github.com/bigdatagenomics/adam/pull/1177) ([heuermh](https://github.com/heuermh))
- Refactor CLIs for merging sharded files [\#1167](https://github.com/bigdatagenomics/adam/pull/1167) ([fnothaft](https://github.com/fnothaft))
- Update Hadoop-BAM to version 7.7.0 [\#1166](https://github.com/bigdatagenomics/adam/pull/1166) ([heuermh](https://github.com/heuermh))
- \[ADAM-1162\] Write record group string name. [\#1163](https://github.com/bigdatagenomics/adam/pull/1163) ([fnothaft](https://github.com/fnothaft))
- Map IntervalList format column four to feature name [\#1159](https://github.com/bigdatagenomics/adam/pull/1159) ([heuermh](https://github.com/heuermh))
- Make AlignmentRecordConverter public so that it can be used from other projects [\#1157](https://github.com/bigdatagenomics/adam/pull/1157) ([tomwhite](https://github.com/tomwhite))
- added predicate option to loadCoverage [\#1156](https://github.com/bigdatagenomics/adam/pull/1156) ([akmorrow13](https://github.com/akmorrow13))
- \[ADAM-1154\] Change set -x to set -e in ./bin/adam-shell. [\#1155](https://github.com/bigdatagenomics/adam/pull/1155) ([fnothaft](https://github.com/fnothaft))
- Remove Gene and related models and parsing code [\#1153](https://github.com/bigdatagenomics/adam/pull/1153) ([heuermh](https://github.com/heuermh))
- Reorder kryo.register statements in ADAMKryoRegistrator [\#1148](https://github.com/bigdatagenomics/adam/pull/1148) ([heuermh](https://github.com/heuermh))
- Updated GenomicPartitioners to accept additional key. [\#1147](https://github.com/bigdatagenomics/adam/pull/1147) ([akmorrow13](https://github.com/akmorrow13))
- \[ADAM-1141\] Add support for saving/loading AlignmentRecords to/from CRAM. [\#1145](https://github.com/bigdatagenomics/adam/pull/1145) ([fnothaft](https://github.com/fnothaft))
- misc pom/test/resource improvements [\#1142](https://github.com/bigdatagenomics/adam/pull/1142) ([ryan-williams](https://github.com/ryan-williams))
- \[ADAM-1136\] Transform runs successfully with kryo registration required [\#1138](https://github.com/bigdatagenomics/adam/pull/1138) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1132\] Fix improper quoting of bash args in adam-shell. [\#1133](https://github.com/bigdatagenomics/adam/pull/1133) ([fnothaft](https://github.com/fnothaft))
- Remove StructuralVariant and StructuralVariantType, add names field to Variant [\#1131](https://github.com/bigdatagenomics/adam/pull/1131) ([heuermh](https://github.com/heuermh))
- Remove StructuralVariant and StructuralVariantType, add names field to Variant [\#1130](https://github.com/bigdatagenomics/adam/pull/1130) ([heuermh](https://github.com/heuermh))
- PR #1108 with issue #1122 [\#1128](https://github.com/bigdatagenomics/adam/pull/1128) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1038\] Eliminate writing to /tmp during CI builds. [\#1127](https://github.com/bigdatagenomics/adam/pull/1127) ([fnothaft](https://github.com/fnothaft))
- Update for bdg-formats code style changes [\#1126](https://github.com/bigdatagenomics/adam/pull/1126) ([heuermh](https://github.com/heuermh))
- \[ADAM-1124\] Add Scoverage and generate coverage reports in Jenkins. [\#1125](https://github.com/bigdatagenomics/adam/pull/1125) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1093\] Move to support Spark 2.0.0. [\#1123](https://github.com/bigdatagenomics/adam/pull/1123) ([fnothaft](https://github.com/fnothaft))
- remove duplicated dependency [\#1119](https://github.com/bigdatagenomics/adam/pull/1119) ([ryan-williams](https://github.com/ryan-williams))
- Clean up ADAMContext [\#1118](https://github.com/bigdatagenomics/adam/pull/1118) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-993\] Support loading files using globs and from directory paths. [\#1117](https://github.com/bigdatagenomics/adam/pull/1117) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1087\] Migrate away from FileSystem.get [\#1116](https://github.com/bigdatagenomics/adam/pull/1116) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1099\] Make reference region not throw NPE. [\#1115](https://github.com/bigdatagenomics/adam/pull/1115) ([fnothaft](https://github.com/fnothaft))
- Add pipes API [\#1114](https://github.com/bigdatagenomics/adam/pull/1114) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1105\] Use assembly jar in adam-shell. [\#1111](https://github.com/bigdatagenomics/adam/pull/1111) ([fnothaft](https://github.com/fnothaft))
- Add outer joins [\#1109](https://github.com/bigdatagenomics/adam/pull/1109) ([fnothaft](https://github.com/fnothaft))
- Modified CalculateDepth to calcuate coverage from alignment files [\#1108](https://github.com/bigdatagenomics/adam/pull/1108) ([akmorrow13](https://github.com/akmorrow13))
- Resolves various single file save/header issues [\#1104](https://github.com/bigdatagenomics/adam/pull/1104) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1100\] Resolve Sample Not Serializable exception [\#1101](https://github.com/bigdatagenomics/adam/pull/1101) ([fnothaft](https://github.com/fnothaft))
- added loadIndexedVcf and loadIndexedBam for multiple ReferenceRegions [\#1096](https://github.com/bigdatagenomics/adam/pull/1096) ([akmorrow13](https://github.com/akmorrow13))
- Added support for Indexed VCF files [\#1095](https://github.com/bigdatagenomics/adam/pull/1095) ([akmorrow13](https://github.com/akmorrow13))
- \[ADAM-582\] Eliminate .get on option in FragmentCoverter. [\#1091](https://github.com/bigdatagenomics/adam/pull/1091) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-776\] Rename duplicate OUTPUT metaVar in ADAM2Fastq. [\#1090](https://github.com/bigdatagenomics/adam/pull/1090) ([fnothaft](https://github.com/fnothaft))
- refactored ReferenceFile to require SequenceDictionary [\#1086](https://github.com/bigdatagenomics/adam/pull/1086) ([akmorrow13](https://github.com/akmorrow13))
- \[ADAM-1073\] Remove network-connected and default test-related Maven profiles [\#1082](https://github.com/bigdatagenomics/adam/pull/1082) ([heuermh](https://github.com/heuermh))
- \[ADAM-1053\] Clean up Transform [\#1081](https://github.com/bigdatagenomics/adam/pull/1081) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1061\] Clean up attributes regex and denormalized fields [\#1080](https://github.com/bigdatagenomics/adam/pull/1080) ([fnothaft](https://github.com/fnothaft))
- Extended TwoBitFile and NucleotideContigFragmentRDDFunctions to behave more similar [\#1079](https://github.com/bigdatagenomics/adam/pull/1079) ([akmorrow13](https://github.com/akmorrow13))
- Refactor variant and genotype annotations [\#1078](https://github.com/bigdatagenomics/adam/pull/1078) ([heuermh](https://github.com/heuermh))
- \[ADAM-1039\] Add basic support for Sample record. [\#1077](https://github.com/bigdatagenomics/adam/pull/1077) ([fnothaft](https://github.com/fnothaft))
- Remove code workarounds necessary for Spark 1.2.1/Hadoop 1.0.x support [\#1076](https://github.com/bigdatagenomics/adam/pull/1076) ([heuermh](https://github.com/heuermh))
- \[ADAM-194\] Use separate filtersFailed and filtersPassed arrays for variant quality filters [\#1075](https://github.com/bigdatagenomics/adam/pull/1075) ([heuermh](https://github.com/heuermh))
- Whitespace code style fixes [\#1074](https://github.com/bigdatagenomics/adam/pull/1074) ([heuermh](https://github.com/heuermh))
- \[ADAM-1006\] Split überjar out to adam-assembly submodule. [\#1072](https://github.com/bigdatagenomics/adam/pull/1072) ([fnothaft](https://github.com/fnothaft))
- Remove code coverage profile [\#1071](https://github.com/bigdatagenomics/adam/pull/1071) ([heuermh](https://github.com/heuermh))
- \[ADAM-768\] ReferenceRegion from variant/genotypes [\#1070](https://github.com/bigdatagenomics/adam/pull/1070) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1044\] Support VCF annotation ANN field [\#1069](https://github.com/bigdatagenomics/adam/pull/1069) ([heuermh](https://github.com/heuermh))
- \[ADAM-1067\] Add release documentation and scripting for Spark Packages. [\#1068](https://github.com/bigdatagenomics/adam/pull/1068) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-602\] Remove plugin code. [\#1065](https://github.com/bigdatagenomics/adam/pull/1065) ([fnothaft](https://github.com/fnothaft))
- Refactoring `org.bdgenomics.adam.io` package. [\#1064](https://github.com/bigdatagenomics/adam/pull/1064) ([fnothaft](https://github.com/fnothaft))
- Cleanup in org.bdgenomics.adam.converters package. [\#1062](https://github.com/bigdatagenomics/adam/pull/1062) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1057\] Remove workaround for gzip/BGZF compressed VCF headers [\#1057](https://github.com/bigdatagenomics/adam/pull/1057) ([heuermh](https://github.com/heuermh))
- Cleanup on `org.bdgenomics.adam.algorithms.smithwaterman` package. [\#1056](https://github.com/bigdatagenomics/adam/pull/1056) ([fnothaft](https://github.com/fnothaft))
- Documentation cleanup and minor refactor on the consensus package. [\#1055](https://github.com/bigdatagenomics/adam/pull/1055) ([fnothaft](https://github.com/fnothaft))
- Add KEYS with public code signing keys [\#1054](https://github.com/bigdatagenomics/adam/pull/1054) ([heuermh](https://github.com/heuermh))
- Adding GA4GH 0.5.1 converter for reads. [\#1052](https://github.com/bigdatagenomics/adam/pull/1052) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-1011\] Refactor to add GenomicRDDs for all Avro types [\#1051](https://github.com/bigdatagenomics/adam/pull/1051) ([fnothaft](https://github.com/fnothaft))
- removed interval trait and redirected to interval in utils-intervalrdd [\#1046](https://github.com/bigdatagenomics/adam/pull/1046) ([akmorrow13](https://github.com/akmorrow13))
- \[ADAM-952\] Expose sorting by reference index. [\#1045](https://github.com/bigdatagenomics/adam/pull/1045) ([fnothaft](https://github.com/fnothaft))
- overlap query reflects new formats [\#1043](https://github.com/bigdatagenomics/adam/pull/1043) ([erictu](https://github.com/erictu))
- Changed loadIndexedBam to use hadoop-bam InputFormat [\#1036](https://github.com/bigdatagenomics/adam/pull/1036) ([fnothaft](https://github.com/fnothaft))
- Increase Avro dependency version to 1.8.0 [\#1034](https://github.com/bigdatagenomics/adam/pull/1034) ([heuermh](https://github.com/heuermh))
- Improved README fix using feedback from other approach review. [\#1034](https://github.com/bigdatagenomics/adam/pull/1034) ([InvisibleTech](https://github.com/InvisibleTech))
- Error in the README.md for kmer.scala example, need to get rdd first. [\#1032](https://github.com/bigdatagenomics/adam/pull/1032) ([InvisibleTech](https://github.com/InvisibleTech))
- Add fragmentEndPosition to NucleotideContigFragment [\#1030](https://github.com/bigdatagenomics/adam/pull/1030) ([heuermh](https://github.com/heuermh))
- Logging to be done by ADAM utils code rather than Spark [\#1028](https://github.com/bigdatagenomics/adam/pull/1028) ([jpdna](https://github.com/jpdna))
- add maxScore [\#1027](https://github.com/bigdatagenomics/adam/pull/1027) ([xubo245](https://github.com/xubo245))
- \[ADAM-1008\] Modify jenkins-test script to support Java 8 build. [\#1026](https://github.com/bigdatagenomics/adam/pull/1026) ([fnothaft](https://github.com/fnothaft))
- whitespace change, do not merge [\#1025](https://github.com/bigdatagenomics/adam/pull/1025) ([shaneknapp](https://github.com/shaneknapp))
- require kryo registration in tests [\#1020](https://github.com/bigdatagenomics/adam/pull/1020) ([ryan-williams](https://github.com/ryan-williams))
- print full stack traces on test failures [\#1019](https://github.com/bigdatagenomics/adam/pull/1019) ([ryan-williams](https://github.com/ryan-williams))
- bump commons-io version [\#1017](https://github.com/bigdatagenomics/adam/pull/1017) ([ryan-williams](https://github.com/ryan-williams))
- exclude javadoc jar in adam-shell [\#1016](https://github.com/bigdatagenomics/adam/pull/1016) ([ryan-williams](https://github.com/ryan-williams))
- \[ADAM-909\] Refactoring variation RDDs. [\#1015](https://github.com/bigdatagenomics/adam/pull/1015) ([fnothaft](https://github.com/fnothaft))
- Modified CalculateDepth to get coverage on whole alignment adam files [\#1010](https://github.com/bigdatagenomics/adam/pull/1010) ([akmorrow13](https://github.com/akmorrow13))
- \[ADAM-1004\] Remove recursive maven.build.timestamp declaration [\#1005](https://github.com/bigdatagenomics/adam/pull/1005) ([heuermh](https://github.com/heuermh))
- Maint 2.11 0.19.0 [\#999](https://github.com/bigdatagenomics/adam/pull/999) ([tushu1232](https://github.com/tushu1232))
- \[ADAM-710\] Add saveAs methods for feature formats GTF, BED, IntervalList, and NarrowPeak [\#998](https://github.com/bigdatagenomics/adam/pull/998) ([heuermh](https://github.com/heuermh))
- Moving Adam2Fastq to ADAM2Fastq [\#995](https://github.com/bigdatagenomics/adam/pull/995) ([heuermh](https://github.com/heuermh))
- Update release doc for CHANGES.md and homebrew [\#994](https://github.com/bigdatagenomics/adam/pull/994) ([heuermh](https://github.com/heuermh))
- Update to AlignmentRecordField and its usages as contig changed to co… [\#992](https://github.com/bigdatagenomics/adam/pull/992) ([jpdna](https://github.com/jpdna))
- \[ADAM-974\] Short term fix for multiple ADAM cli assembly jars check [\#990](https://github.com/bigdatagenomics/adam/pull/990) ([heuermh](https://github.com/heuermh))
- Update hadoop-bam dependency version to 7.5.0 [\#989](https://github.com/bigdatagenomics/adam/pull/989) ([heuermh](https://github.com/heuermh))
- Replaced Contig with ContigName in AlignmentRecord and related changes [\#988](https://github.com/bigdatagenomics/adam/pull/988) ([jpdna](https://github.com/jpdna))
- fix some deprecation/style things and rename a pkg [\#986](https://github.com/bigdatagenomics/adam/pull/986) ([ryan-williams](https://github.com/ryan-williams))
- Fix Adam2fastq in case of read with both reverse and unmapped flags [\#982](https://github.com/bigdatagenomics/adam/pull/982) ([jpdna](https://github.com/jpdna))
- \[ADAM-510\] Refactoring RDD function names [\#979](https://github.com/bigdatagenomics/adam/pull/979) ([heuermh](https://github.com/heuermh))
- Use .adam/_{seq,rg}dict.avro paths for Avro-formatted dictionaries [\#978](https://github.com/bigdatagenomics/adam/pull/978) ([heuermh](https://github.com/heuermh))
- Remove unused file VcfHeaderUtils.scala [\#977](https://github.com/bigdatagenomics/adam/pull/977) ([heuermh](https://github.com/heuermh))
- add validation stringency to bam parsing, flagstat [\#976](https://github.com/bigdatagenomics/adam/pull/976) ([ryan-williams](https://github.com/ryan-williams))
- more permissible jar regex in adam-submit [\#975](https://github.com/bigdatagenomics/adam/pull/975) ([ryan-williams](https://github.com/ryan-williams))
- fix bash arg array processing in adam-submit [\#972](https://github.com/bigdatagenomics/adam/pull/972) ([ryan-williams](https://github.com/ryan-williams))
- adamGetReferenceString reduces pairs correctly, fixes #967 [\#970](https://github.com/bigdatagenomics/adam/pull/970) ([erictu](https://github.com/erictu))
- A few improvements [\#966](https://github.com/bigdatagenomics/adam/pull/966) ([ryan-williams](https://github.com/ryan-williams))
- improve SW performance by replacing functional reductions with imperative ones [\#965](https://github.com/bigdatagenomics/adam/pull/965) ([noamBarkai](https://github.com/noamBarkai))
- \[ADAM-962\] Fix corrupt single-file BAM output. [\#964](https://github.com/bigdatagenomics/adam/pull/964) ([fnothaft](https://github.com/fnothaft))
- \[ADAM-960\] Updating bdg-utils dependency version to 0.2.4 [\#961](https://github.com/bigdatagenomics/adam/pull/961) ([heuermh](https://github.com/heuermh))
- \[ADAM-946\] Fixes to FlagStat for Samtools concordance issue [\#954](https://github.com/bigdatagenomics/adam/pull/954) ([jpdna](https://github.com/jpdna))
- Use hadoop-bam BAMInputFormat to do loadIndexedBam [\#953](https://github.com/bigdatagenomics/adam/pull/953) ([andrewmchen](https://github.com/andrewmchen))
- Add -print_metrics option to Jenkins build [\#947](https://github.com/bigdatagenomics/adam/pull/947) ([heuermh](https://github.com/heuermh))
- adam2vcf doesn't have info fields [\#939](https://github.com/bigdatagenomics/adam/pull/939) ([andrewmchen](https://github.com/andrewmchen))
- \[ADAM-893\] Register missing serializers. [\#933](https://github.com/bigdatagenomics/adam/pull/933) ([fnothaft](https://github.com/fnothaft))

