<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Variant Annotation Proposal from Mount Sinai - Big Data Genomics</title>
  <meta name="author" content="Big Data Genomics">

  
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://bigdatagenomics.github.io/rfc/3">
  <link href="/favicon.png" type="image/png" rel="icon">
  <link href="/atom.xml" rel="alternate" title="Big Data Genomics" type="application/atom+xml">

  <link href="/assets/bootstrap/dist/css/bootstrap.min.css" rel="stylesheet" type="text/css">
<link href="/assets/bootstrap/dist/css/bootstrap-theme.min.css" rel="stylesheet" type="text/css">
<link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">


  <script src="/javascripts/libs/jquery/jquery-2.0.3.min.js"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-49914824-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

  <body   >
    <div id="wrap">
      <header role="banner">
        <nav class="navbar navbar-default" role="navigation">
    <div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="/">Big Data Genomics</a>
        </div>

        <div class="navbar-collapse collapse">
            <ul class="nav navbar-nav">
                <li >
                    <a href="/">Blog</a>
                </li>
                <li >
                    <a href="/blog/archives">Archives</a>
                </li>
                <li >
                    <a href="/projects">Projects</a>
                </li>
                <li >
                    <a href="/mail">Mailing List</a>
                </li>
                <li >
                    <a href="/cla">CLAs</a>
                </li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a class="subscribe-rss" href="/atom.xml" title="subscribe via RSS">
                        <span class="visible-xs">RSS</span>
                        <img class="hidden-xs" src="/images/rss.png" alt="RSS">
                    </a>
                </li>
                
            </ul>
            
                <form class="search navbar-form navbar-right" action="http://google.com/search" method="GET">
                    <input type="hidden" name="q" value="site:bigdatagenomics.github.io">
                    <div class="form-group">
                        <input class="form-control" type="text" name="q" placeholder="Search">
                    </div>
                </form>
            
        </div>
    </div>
</nav>


      </header>
      <div id="main" class="container">
        <div id="content">
	       <img src="/images/github.png" /><a href="https://github.com/bigdatagenomics/">https://github.com/bigdatagenomics/</a>&nbsp;&nbsp;&nbsp;&nbsp;<img src="/images/twitter.png"/><a href="https://twitter.com/bigdatagenomics">https://twitter.com/bigdatagenomics/</a><br/>
          <div class="row">
  <div class="page-content col-md-9">
    <article role="article">
      
        <header class="page-header">
          
            <p class="meta text-muted text-uppercase">












<span class="glyphicon glyphicon-calendar"></span> <time datetime="2014-01-29T16:08:00-08:00" pubdate data-updated="true">Jan 29<span>th</span>, 2014</time></p>
          
          
            <h1 class="entry-title">
              Variant Annotation Proposal From Mount Sinai
              
            </h1>
          
        </header>
      
      <div class="entry-content">
        <p>Status: DRAFT</p>

<h3>Introduction</h3>

<p>Similar to <a href="/rfc/2">RFC #2</a>, we&rsquo;d like to propose one way of storing
annotations in a format that is amenable to joining against ADAM
variant data.  The purpose of this RFC is to propose a schema (in the
form of AVRO IDL) to represent both a set of called variants and
annotations on those variants.</p>

<p>One particular requirement of our approach is to separate several concepts:</p>

<ul>
<li><p>The notion of what a variant is, which is represented by the ADAMVariant record:</p>

<ul>
<li>Chromosome</li>
<li>Location on the chromosome</li>
<li>Reference allele</li>
<li>Alternate allele</li>
</ul>


<p>We believe that by making a minimalist definition of a variant, and using that
as a key in the various other structures, a high degree of consistency is achieved with respect
to what annotations and genotypes are actually associated with.</p></li>
<li><p>A set of annotations on a variant, such as dbSNP ID, Clinvar data,
or arbitrary properties of a novel variant, represented by
ADAMDatabaseVariantAnnotation.  In this proposal, it only contains the
dbSNP ID, but, in the future, it will contain a field for each new
type of annotation.</p></li>
<li><p>A set of genotypes that use the variant definition (1) to indicate
calls on samples, as represented by the ADAMGenotype record.</p></li>
</ul>


<h3>Schema</h3>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
<span class='line-number'>14</span>
<span class='line-number'>15</span>
<span class='line-number'>16</span>
<span class='line-number'>17</span>
<span class='line-number'>18</span>
<span class='line-number'>19</span>
<span class='line-number'>20</span>
<span class='line-number'>21</span>
<span class='line-number'>22</span>
<span class='line-number'>23</span>
<span class='line-number'>24</span>
<span class='line-number'>25</span>
<span class='line-number'>26</span>
<span class='line-number'>27</span>
<span class='line-number'>28</span>
<span class='line-number'>29</span>
<span class='line-number'>30</span>
<span class='line-number'>31</span>
<span class='line-number'>32</span>
<span class='line-number'>33</span>
<span class='line-number'>34</span>
<span class='line-number'>35</span>
<span class='line-number'>36</span>
<span class='line-number'>37</span>
<span class='line-number'>38</span>
<span class='line-number'>39</span>
<span class='line-number'>40</span>
<span class='line-number'>41</span>
<span class='line-number'>42</span>
<span class='line-number'>43</span>
<span class='line-number'>44</span>
<span class='line-number'>45</span>
<span class='line-number'>46</span>
<span class='line-number'>47</span>
<span class='line-number'>48</span>
<span class='line-number'>49</span>
<span class='line-number'>50</span>
<span class='line-number'>51</span>
<span class='line-number'>52</span>
<span class='line-number'>53</span>
<span class='line-number'>54</span>
<span class='line-number'>55</span>
<span class='line-number'>56</span>
<span class='line-number'>57</span>
<span class='line-number'>58</span>
<span class='line-number'>59</span>
<span class='line-number'>60</span>
<span class='line-number'>61</span>
<span class='line-number'>62</span>
<span class='line-number'>63</span>
<span class='line-number'>64</span>
<span class='line-number'>65</span>
<span class='line-number'>66</span>
<span class='line-number'>67</span>
<span class='line-number'>68</span>
<span class='line-number'>69</span>
<span class='line-number'>70</span>
<span class='line-number'>71</span>
<span class='line-number'>72</span>
<span class='line-number'>73</span>
<span class='line-number'>74</span>
<span class='line-number'>75</span>
<span class='line-number'>76</span>
<span class='line-number'>77</span>
<span class='line-number'>78</span>
<span class='line-number'>79</span>
<span class='line-number'>80</span>
<span class='line-number'>81</span>
<span class='line-number'>82</span>
<span class='line-number'>83</span>
<span class='line-number'>84</span>
<span class='line-number'>85</span>
<span class='line-number'>86</span>
<span class='line-number'>87</span>
<span class='line-number'>88</span>
<span class='line-number'>89</span>
</pre></td><td class='code'><pre><code class='c'><span class='line'><span class="n">protocol</span> <span class="n">ADAM</span> <span class="p">{</span>
</span><span class='line'><span class="n">import</span> <span class="n">idl</span> <span class="s">&quot;adam.avdl&quot;</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'><span class="n">record</span> <span class="n">ADAMContig</span> <span class="p">{</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">string</span> <span class="p">}</span> <span class="n">contigName</span><span class="p">;</span>
</span><span class='line'>  <span class="c1">// Do we need the reference filename if we also have URL?</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">string</span> <span class="p">}</span> <span class="n">referenceName</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">long</span> <span class="p">}</span>   <span class="n">referenceLength</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">string</span> <span class="p">}</span> <span class="n">referenceURL</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">string</span> <span class="p">}</span> <span class="n">referenceMD5</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'>
</span><span class='line'><span class="n">record</span> <span class="n">ADAMVariant</span> <span class="p">{</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">ADAMContig</span> <span class="p">}</span> <span class="n">contig</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">long</span> <span class="p">}</span>       <span class="n">position</span><span class="p">;</span>
</span><span class='line'>  <span class="n">array</span><span class="o">&lt;</span><span class="n">Base</span><span class="o">&gt;</span> <span class="n">referenceAlleles</span><span class="p">;</span>
</span><span class='line'>  <span class="n">array</span><span class="o">&lt;</span><span class="n">Base</span><span class="o">&gt;</span> <span class="n">variantAlleles</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'>
</span><span class='line'><span class="k">enum</span> <span class="n">ADAMGenotypeAllele</span> <span class="p">{</span>
</span><span class='line'>  <span class="n">Ref</span><span class="p">,</span> <span class="n">Alt</span><span class="p">,</span> <span class="n">NoCall</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'>
</span><span class='line'><span class="c1">// This record represents all stats that, inside a VCF, are stored outside of the</span>
</span><span class='line'><span class="c1">// sample but are computed based on the samples.  For instance,  MAPQ0 is an aggregate</span>
</span><span class='line'><span class="c1">// stat computed from all samples and stored inside the INFO line.</span>
</span><span class='line'><span class="n">record</span> <span class="n">VariantStatistics</span> <span class="p">{</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">double</span> <span class="p">}</span> <span class="n">baseQRankSum</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">double</span> <span class="p">}</span> <span class="n">clippingRankSum</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">int</span> <span class="p">}</span> <span class="n">readDepth</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">boolean</span> <span class="p">}</span> <span class="n">downsampled</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">double</span> <span class="p">}</span> <span class="n">fisherStrandBiasPValue</span><span class="p">;</span> <span class="c1">// Phred-scaled.</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">float</span> <span class="p">}</span> <span class="n">haplotypeScore</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">float</span> <span class="p">}</span> <span class="n">inbreedingCoefficient</span><span class="p">;</span>
</span><span class='line'>  <span class="n">array</span><span class="o">&lt;</span><span class="kt">int</span><span class="o">&gt;</span> <span class="n">alleleCountMLE</span><span class="p">;</span>
</span><span class='line'>  <span class="n">array</span><span class="o">&lt;</span><span class="kt">int</span><span class="o">&gt;</span> <span class="n">alleleFrequencyMLE</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">float</span> <span class="p">}</span> <span class="n">rmsMapQ</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">int</span> <span class="p">}</span> <span class="n">mapq0reads</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">float</span> <span class="p">}</span> <span class="n">mqranksum</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">boolean</span> <span class="p">}</span> <span class="n">usedForNegativeTrainingSet</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">boolean</span> <span class="p">}</span> <span class="n">usedForPositiveTrainingSet</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">float</span> <span class="p">}</span> <span class="n">variantQualityByDepth</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">float</span> <span class="p">}</span> <span class="n">readPositionRankSum</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">float</span> <span class="p">}</span> <span class="n">vqslod</span><span class="p">;</span> <span class="c1">// log odds ratio of being a true vs false variant under trained gaussian mixture model</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">string</span> <span class="p">}</span> <span class="n">culprit</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'>
</span><span class='line'><span class="n">record</span> <span class="n">ADAMGenotype</span> <span class="p">{</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">ADAMVariant</span> <span class="p">}</span> <span class="n">variant</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'>  <span class="c1">// Variant-level &quot;provenance&quot; data, i.e. data shared amongst all concurrently genotyped samples</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">VariantStatistics</span> <span class="p">}</span> <span class="n">variantStats</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">boolean</span> <span class="p">}</span> <span class="n">varIsFiltered</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>  <span class="c1">// &quot;null&quot; implies no filters were applied</span>
</span><span class='line'>  <span class="n">array</span> <span class="o">&lt;</span><span class="n">string</span><span class="o">&gt;</span> <span class="n">varFilters</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'>  <span class="c1">// Sample-level data, i.e. data specific to this particular sample</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">string</span> <span class="p">}</span> <span class="n">sampleId</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'>  <span class="c1">// Length is equal to the ploidy</span>
</span><span class='line'>  <span class="n">array</span> <span class="o">&lt;</span><span class="n">ADAMGenotypeAllele</span><span class="o">&gt;</span> <span class="n">alleles</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">int</span> <span class="p">}</span> <span class="n">referenceReadDepth</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">int</span> <span class="p">}</span> <span class="n">alternateReadDepth</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">int</span> <span class="p">}</span> <span class="n">readDepth</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">string</span> <span class="p">}</span> <span class="n">filter</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">int</span> <span class="p">}</span> <span class="n">genotypeQuality</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'>  <span class="c1">// In the ADAM world we split multiallelic VCF lines into multiple</span>
</span><span class='line'>  <span class="c1">// single-alternate records.  This bit is set if that happened for this</span>
</span><span class='line'>  <span class="c1">// record.</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">boolean</span> <span class="p">}</span> <span class="n">splitFromMultiAllelic</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'>  <span class="n">array</span><span class="o">&lt;</span><span class="kt">int</span><span class="o">&gt;</span> <span class="n">genotypeLikelihoods</span><span class="p">;</span> <span class="c1">// Phred-scaled. Always length 3 since we are not multiallelic.</span>
</span><span class='line'>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">boolean</span> <span class="p">}</span> <span class="n">isPhased</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">string</span> <span class="p">}</span>  <span class="n">phaseSetId</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">int</span> <span class="p">}</span>     <span class="n">phaseQuality</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'>
</span><span class='line'><span class="n">record</span> <span class="n">ADAMDatabaseVariantAnnotation</span> <span class="p">{</span>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="n">ADAMVariant</span> <span class="p">}</span> <span class="n">variant</span><span class="p">;</span>
</span><span class='line'>
</span><span class='line'>  <span class="k">union</span> <span class="p">{</span> <span class="n">null</span><span class="p">,</span> <span class="kt">int</span> <span class="p">}</span> <span class="n">dbsnpId</span> <span class="o">=</span> <span class="n">null</span><span class="p">;</span>
</span><span class='line'><span class="p">}</span>
</span><span class='line'>
</span><span class='line'><span class="p">}</span>
</span></code></pre></td></tr></table></div></figure>




      </div>
      
        <footer>
          <p class="meta text-muted">
            
            












<span class="glyphicon glyphicon-calendar"></span> <time datetime="2014-01-29T16:08:00-08:00" pubdate data-updated="true">Jan 29<span>th</span>, 2014</time>
            
          </p>
          
            <div class="sharing">
  
  
  
    <div class="fb-like" data-send="true" data-width="450" data-show-faces="false"></div>
  
</div>

          
        </footer>
      
    </article>
    
      <section>
        <h1>Comments</h1>
        <div id="disqus_thread" aria-live="polite"><noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
      </section>
    
  </div>

  
    <aside class="sidebar col-md-3">
      
        <a href="/">
<img src="https://avatars2.githubusercontent.com/u/5839438?s=256&v=4" height="256" width="256"/>
</a>

<hr size="1"/>
<p><font size="+1">About us...</font></p>
<p><em>
This project is supported in part by NIH BD2K Award 1-U54HG007990-01 and NIH Cancer Cloud Pilot Award HHSN261201400006C with collaborators from the <a href="https://amplab.cs.berkeley.edu/">AMPLab</a> at UC Berkeley, <a href="https://genomics.soe.ucsc.edu/haussler/cancer_genomics">Genome Informatics Lab</a> at UC Santa Cruz, <a href="http://icahn.mssm.edu/">Icahn School of Medicine</a> at Mount Sinai, <a href="http://research.microsoft.com/">Microsoft Research</a>, <a href="http://www.cloudera.com/">Cloudera</a>, and the <a href="https://www.broadinstitute.org/">Broad Institute</a>.</em></p>

<hr size="1"/>
<p>
<em>Chat with us...</em>
<a href="https://gitter.im/bigdatagenomics/adam?utm_source=badge&amp;utm_medium=badge&amp;utm_campaign=pr-badge"><img src="https://badges.gitter.im/Join%20Chat.svg" alt="Gitter" /></a>
</p>
<p>
If you're interested in contributing, take a look at the open <a href="https://github.com/bigdatagenomics/adam/issues?labels=pick+me+up%21&page=1&state=open">"pick me up!" issues</a>.
</p>
<hr size="1"/>


<section class="panel panel-default">
  <div class="panel-heading">
    <h3 class="panel-title">Recent Posts</h3>
  </div>
  
  <div id="recent_posts" class="list-group">
    
    <a class="list-group-item " href="/blog/2018/03/28/adam-0-dot-24-dot-0-cannoli-0-dot-2-dot-0-releases/">ADAM 0.24.0 and Cannoli 0.2.0 Released</a>
    
    <a class="list-group-item " href="/blog/2018/01/04/adam-0-dot-23-dot-0-released-plus-avocado-cannoli-and-deca-releases/">ADAM 0.23.0 Released (+ Avocado and DECA Releases)</a>
    
    <a class="list-group-item " href="/blog/2017/04/03/adam-0-dot-22-dot-0-release/">ADAM 0.22.0 Released</a>
    
    <a class="list-group-item " href="/blog/2017/01/06/adam-0-dot-21-dot-0-release/">ADAM 0.21.0 Released</a>
    
    <a class="list-group-item " href="/blog/2016/10/19/adam-0-dot-20-dot-0-release/">ADAM 0.20.0 Released</a>
    
  </div>
</section>
<font size="-1">
YourKit is supporting the Big Data Genomics open source project with its full-featured Java Profiler.
YourKit, LLC is the creator of innovative and intelligent tools for profiling
Java and .NET applications. Take a look at YourKit's leading software products:
<a href="http://www.yourkit.com/java/profiler/index.jsp">YourKit Java Profiler</a> and
<a href="http://www.yourkit.com/.net/profiler/index.jsp">YourKit .NET Profiler</a>.
</font>

      
    </aside>
  
</div>

        </div>
      </div>
    </div>
    <footer role="contentinfo"><div class="container">
    <p class="text-muted credits">
  Copyright &copy; 2018 - Big Data Genomics<br>
  <small>
      <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>,
      <span class="credit">customized with <a href="https://github.com/kAworu/octostrap3">octostrap3</a></span>.
  </small>
</p>

</div>
</footer>
    <script src="/assets/bootstrap/dist/js/bootstrap.min.js"></script>
<script src="/javascripts/modernizr-2.0.js"></script>


<script type="text/javascript">
      var disqus_shortname = 'bigdatagenomics';
      
        
        // var disqus_developer = 1;
        var disqus_identifier = 'http://bigdatagenomics.github.io/rfc/3/index.html';
        var disqus_url = 'http://bigdatagenomics.github.io/rfc/3/index.html';
        var disqus_script = 'embed.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>



<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id; js.async = true;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>









  </body>
</html>
